packages/lit-html/src/directive-helpers.ts

```diff
@@ -0,0 +1,196 @@
+ * @param part Part to insert, or undefined to create a new part
+ * @param refPart Part before which to add the new NodePart; when omitted the
+ *     part added to the end of the `containerPart`
+ */
+export const insertPart = (
+  containerPart: NodePart,
+  part: NodePart | undefined,
+  refPart?: NodePart
+): NodePart => {
+  const container = containerPart._$startNode.parentNode!;
+
+  const refNode =
+    refPart === undefined ? containerPart._$endNode : refPart._$startNode;
+
+  if (part === undefined) {
+    const startNode = container.insertBefore(createMarker(), refNode);
+    const endNode = container.insertBefore(createMarker(), refNode);
+    part = new NodePart(
+      startNode,
+      endNode,
+      containerPart,
+      containerPart.options
+    );
+  } else {
+    const endNode = part._$endNode!.nextSibling;
+    if (endNode !== refNode) {
+      let start: Node | null = part._$startNode;
+      while (start !== endNode) {
+        const n: Node | null = start!.nextSibling;
+        container.insertBefore(start!, refNode);
+        start = n;
+      }
+    }
+  }
+
+  return part;
+};
+
+/**
+ * Sets the value of a Part.
+ *
+ * Note that this should only be used to set/update the value of user-created
+ * parts (i.e. those created using `createAndInsertPart`); it should not be used
+ * by directives to set the value of the directive's container part. Directives
+ * should return a value from `update`/`render` to update their part state.
+ *
+ * For directives that require setting their part value asynchronously, they
+ * should extend `DisconnectableDirective` and call `this.setValue()`.
+ *
+ * @param part Part to set
+ * @param value Value to set
+ * @param index For `AttributePart`s, the index to set
+ * @param directiveParent Used internally; should not be set by user
+ */
+export const setPartValue = <T extends Part>(
+  part: T,
+  value: unknown,
+  index?: number,
+  directiveParent: DirectiveParent = part
+): T => {
+  if ((part as AttributePart).strings !== undefined) {
+    if (index === undefined) {
+      throw new Error(
+        "An index must be provided to set an AttributePart's value."
+      );
+    }
+    const newValues = [...(part._$value as Array<unknown>)];
+    newValues[index] = value;
+    (part as AttributePart)._$setValue(newValues, directiveParent, 0);
+  } else {
+    part._$setValue(value, directiveParent);
+  }
+  return part;
+};
+
+// A sentinal value that can never appear as a part value except when set by
+// live(). Used to force a dirty-check to fail and cause a re-render.
+const RESET_VALUE = {};
+
+/**
+ * Resets the stored state of a NodePart used for dirty-checking and
+ * efficiently updating the part to the given value, or when omitted, to a state
+ * where any subsequent dirty-check is guaranteed to fail, causing the next
+ * commit to take effect.
+ *
+ * @param part
+ * @param value
+ */
+export const resetPartValue = (part: Part, value: unknown = RESET_VALUE) =>
```

Comment:
We resolved to call this `committedValue`, right? And then this would be `resetPartCommittedValue` and `getPartCommittedValue`? (Can handler later)

Reply:
I didn't make this change yet